staging/src/k8s.io/apiserver/pkg/util/flag/map_string_bool_test.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flag
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
```

Comment:
I really dislike assert libraries. I want to spend a week and remove these from our dependencies. Generic assert helpers are against the internal style guide. https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures

Reply:
> It may be tempting to write a bunch of assertFoo helpers, but be sure your helpers produce useful error messages.

It doesn't sound against the style guide to me, as long as the assert helpers produce useful errors. I'm easy either way though.

Reply:
They are against the internal style guide, not the external one. This wouldn't get past a go readability review.

Reply:
@mikedanese I would strongly prefer keeping them. These helpers make writing test code so much cleaner (as long as the error messages are useful).

Reply:
All of the assertions in this test can be written in plain go with equivalent LOC. Golang has fine support for everything here. What value does a seperate testing sublanguage add? You inevitably duplicate a bunch of features to write imprecise tests that omit useful information from error messages.

https://golang.org/doc/faq#testing_framework

Reply:
Compare something that's fairly common in our codebase:
```go
if e, a := "someExpectedValue", someObj.someField; e != a {
  t.Errorf("someField: expected %v, got %v", e, a)
}
```
with
```go
assert.Equal(t, "someExpectedValue", someObj.someField, "unexpected someField value")
```

I find the latter nicer, since it's 1 line instead of 3. How do you do that in 1 without an assertion library?

Reply:
I am not going to fight tooth and nail over this - there are more important things to argue over 😄 . Given that some tests do use testify, and others don't, I think this is probably something that needs to be decided by more than just 1 or 2 of us.